eth/gasprice/gasprice.go

```diff
@@ -201,18 +203,66 @@ func (gpo *Oracle) getBlockPrices(ctx context.Context, signer types.Signer, bloc
 	for _, tx := range txs {
 		sender, err := types.Sender(signer, tx)
 		if err == nil && sender != block.Coinbase() {
-			prices = append(prices, tx.GasPrice())
+			// Remove transactions with a gasPrice of 0
+			if tx.GasPrice() != big.NewInt(0) {
+				prices = append(prices, tx.GasPrice())
+			}
 			if len(prices) >= limit {
 				break
 			}
 		}
 	}
+	if len(prices) > 0 {
+		prices = removeOutliers(prices)
+	}
 	select {
 	case result <- getBlockPricesResult{prices, nil}:
 	case <-quit:
 	}
 }
 
+// removeOutliers calculates the interquartile range of gas prices that are
+// significant outliers with the goal of removing edge cases where MEV skews
+// the gas prices
+func removeOutliers(prices []*big.Int) []*big.Int {
```

Comment:
I think it's sufficient to filter out `0` and `1`, I don't see any need for this method. I might be convinced otherwise if you have some good arguments for why this is needed.